### PR TITLE
chore(chart): Add XYAxis interface for publish properties

### PR DESCRIPTION
**Pull request by jbrundage**
pointSizeColumn is now optional
Fixed an issue with yAxisTypeLogBase

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
I created a diff of the output of this code snippet before and after this PR's code change.
```javascript
import { XYAxis } from "@hpcc-js/chart";
const a = new XYAxis();
console.log(JSON.stringify(a._publishedProperties, null, "  "))
```
Here's the diff: https://gist.github.com/jbrundage/de8d813f629f0da02fd8fce661f4d829/revisions?diff=unified#diff-90926d6e3c250de42124b3f713fed614

### CODE REVIEW

File: packages/chart/src/XYAxis.ts [hunk 1]
```diff
@@ -640,93 +640,127 @@ export class XYAxis extends SVGWidget {
...
+    layers(): XYAxis[];
+    layers(_: XYAxis[]): this;
+    xAxisType_default(_: string);
+    yAxisType_default(_: string);
+    orientation_default(_: string);
+    xAxisHidden_default(_: boolean);
+    yAxisHidden_default(_: boolean);
+    xAxisGuideLines_default(_: boolean);
+    yAxisGuideLines_default(_: boolean);
+}
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Will you add the correct types at the same time?  e.g. `orientation: "horizontal"|"vertical"`

Comment (jbrundage) [reply packages/chart/src/XYAxis.ts]: @GordonSmith added

Comment (jbrundage) [reply packages/chart/src/XYAxis.ts]: `orientation` was the only one on XYAxis that needs that type of change, right?

File: packages/chart/src/XYAxis.ts [hunk 2]
```diff
@@ -640,93 +640,130 @@ export class XYAxis extends SVGWidget {
...
+    selectionMode(): boolean;
+    selectionMode(_: boolean): this;
+    xAxisTitle(): string;
+    xAxisTitle(_: string): this;
+    xAxisTickCount(): number;
+    xAxisTickCount(_: number): this;
+    xAxisTickFormat(): string;
+    xAxisTickFormat(_: string): this;
+    xAxisType(): string;
+    xAxisType(_: string): this;
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Should have better types

File: packages/chart/src/XYAxis.ts [hunk 3]
```diff
@@ -640,93 +640,130 @@ export class XYAxis extends SVGWidget {
...
+    xAxisDomainLow(): number | string;
+    xAxisDomainLow(_: number | string): this;
+    xAxisDomainHigh(): number | string;
+    xAxisDomainHigh(_: number | string): this;
+    xAxisOrdinalPaddingInner(): number;
+    xAxisOrdinalPaddingInner(_: number): this;
+    xAxisOrdinalPaddingOuter(): number;
+    xAxisOrdinalPaddingOuter(_: number): this;
+    xAxisOverlapMode(): string;
+    xAxisOverlapMode(_: string): this;
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Should have better types

File: packages/chart/src/XYAxis.ts [hunk 4]
```diff
@@ -640,93 +640,130 @@ export class XYAxis extends SVGWidget {
...
+    xAxisHidden(): boolean;
+    xAxisHidden(_: boolean): this;
+    yAxisTitle(): string;
+    yAxisTitle(_: string): this;
+    yAxisTickCount(): number;
+    yAxisTickCount(_: number): this;
+    yAxisTickFormat(): string;
+    yAxisTickFormat(_: string): this;
+    yAxisType(): string;
+    yAxisType(_: string): this;
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Should have better types

File: packages/chart/src/XYAxis.ts [hunk 5]
```diff
@@ -640,93 +640,130 @@ export class XYAxis extends SVGWidget {
...
+    regions(_: object[]): this;
+    layers(): XYAxis[];
+    layers(_: XYAxis[]): this;
+    xAxisType_default(_: string);
+    yAxisType_default(_: string);
+    orientation_default(_: string);
+    xAxisHidden_default(_: boolean);
+    yAxisHidden_default(_: boolean);
+    xAxisGuideLines_default(_: boolean);
+    yAxisGuideLines_default(_: boolean);
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: Move the `_defaults` so they are beside their corresponding methods

**Comment by jbrundage** [comment]
@GordonSmith done

**Comment by jbrundage** [comment]
@GordonSmith something in comms is causeing TravisCI to fail

**Pull request merged**